libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1454,8 +1454,41 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			if (equal)
 				body = "converted := value";
 			else
-				solUnimplementedAssert(false, "Array conversion not implemented.");
+			{
+				ArrayType const& from = dynamic_cast<decltype(from)>(_from);
+				ArrayType const& to = dynamic_cast<decltype(to)>(_to);
 
+				//solUnimplementedAssert(false, "Array conversion not implemented.");
+				switch (to.location())
+				{
+				case DataLocation::Storage:
+					// Other cases are done explicitly in LValue::storeValue, and only possible by assignment.
+					solAssert(
+						(to.isPointer() || (from.isByteArray() && to.isByteArray())) &&
+						to.location() == DataLocation::Storage,
+						"Invalid conversion to storage type."
+					);
+					body = "converted := value";
+					break;
+				case DataLocation::Memory:
+					// Copy the array to a free position in memory, unless it is already in memory.
+					if (from.location() != DataLocation::Memory)
+					{
+						solAssert(false, "TODO");
+					}
+					body = "converted := value";
+					break;
+				case DataLocation::CallData:
+					solAssert(
+						to.isByteArray() &&
+						from.isByteArray() &&
+						from.location() == DataLocation::CallData,
```

Comment:
This probably needs more than one stack slot. Maybe leave that for later.

Reply:
This is part of a single commit, that has to be split out into its own PR anyways. I remember I did that because I needed (at least) the functionality to convert from `bytes memory` to `string memory`.